File: octave/spline3_test.m
% Cubic spline interpolation test

nq = 100;   % query points
n  = 5; %15;    % number of interpolation nodes

% interpolation nodes
x = linspace(-1, 1, n);
y = 1 ./ (1 + 25*x.^2);

% Get function values over 100 evenly spaced values
X = linspace(min(x), max(x), nq);
f = 1 ./ (1 + 25*X.^2);

% Get yy values via interpolation
Y = spline3(x, y, X);

% pre-computed y values are plotted with 'o',
% interpolated y values are plotted with '.'
plot(X, Y, '+', X, f, x, y, 'o');

% Wait for input
pause;

File: python/spline3_test.py
# Cubic spline interpolation test

from spline3 import spline3
import numpy as np
import matplotlib.pyplot as plt

nq = 100    # query points
n  = 15     # interpolation nodes

x = np.linspace(-1, 1, n)
y = np.array([ 1.0 / (1 + 25*x[i]**2) for i in range(0, n) ])

# nq evenly spaced values between min(x) and max(x)
X = np.linspace(min(x), max(x), nq)
# Get true function values at the query points
f = [ 1.0 / (1 + 25*X[i]**2) for i in range(0, nq) ]

# Get Y values at query points using cubic spline interpolation
Y = spline3(x, y, X)

# Plot the result
plt.scatter(X, Y, linewidths=0.2, marker='+')
plt.plot(X, f)
plt.scatter(x, y, linewidths=0.2, marker='o')
plt.show()
